include/nlohmann/json.hpp

```diff
@@ -4131,13 +4134,16 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     /// @brief generate SAX events
     /// @sa https://json.nlohmann.me/api/basic_json/sax_parse/
     template<class IteratorType, class SAX>
-    JSON_HEDLEY_NON_NULL(3)
```

Comment:
Please keep these annotations, but suppress Clang's `Wtautological-pointer-compare` warning via

```cpp
#if defined(__clang__)
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wtautological-pointer-compare"
#endif
```

and

```cpp
#if defined(__clang__)
    #pragma clang diagnostic pop
#endif
```